blockchain/src/blockchain/blockchain.rs

```diff
@@ -69,13 +70,17 @@ pub struct BlockchainConfig {
     /// Maximum number of epochs (other than the current one) that the ChainStore will store fully.
     /// Epochs older than this number will be pruned.
     pub max_epochs_stored: u32,
+    /// The history store that is used by the full blockchain.
+    /// If this is set to true, the light history store is used.
+    pub light_history_store: bool,
```

Comment:
AFAIK, `keep_history` is used to determine if we want to prune the history store,  keeping only up to "max_epochs_stored" epochs, whereas "light_history_store" specifies the type of History Store that is being used, and these two are independent.
i.e.: you could potentially have a node using a light story store where keep_history is true  (not sure if there is a  use case for this)
Also, with this implementation, it is possible to have full nodes with the regular history store and full nodes with the light history store.

However, if we decide that:

- Full nodes only use the light story store and they only keep up to max_epochs_stored 
 
Then this can indeed be simplified, as the different combinations mentioned above would no longer be supported